documentation/DCP-documentation/step_2_submit_jobs.md

```diff
@@ -38,18 +38,77 @@ Job files that don't include it will use the default structure.
     This approach is best when you have a large number of groups and the group structure is uniform.
 
       Example: for a 96-well plate experiment where one there are 3 plates and the experiment is grouped by Plate and Well, `batches.sh` would read:
-        `parallel echo '{\"Metadata\": \"Metadata_Plate={1},Metadata_Well={2}{3}\"},' ::: Plate1 Plate2 Plate3 ::: A B C D E F G H ::: 01 02 03 04 05 06 07 08 09 10 11 12 | sort > batches.txt`      
-    * You may also use the list of groupings created by calling `cellprofiler --print-groups` from the command line (see [here](https://github.com/CellProfiler/CellProfiler/wiki/Adapting-CellProfiler-to-a-LIMS-environment#cmd) and [here](https://github.com/CellProfiler/Distributed-CellProfiler/issues/52) for more information).  
-    Note that for job files that specify groupings in this way, the `output_structure` variable is NOT optional - it must be specified or an error will be returned.
+        `parallel echo '{\"Metadata\": \"Metadata_Plate={1},Metadata_Well={2}{3}\"},' ::: Plate1 Plate2 Plate3 ::: A B C D E F G H ::: 01 02 03 04 05 06 07 08 09 10 11 12 | sort > batches.txt`
+  * You may also use the list of groupings created by calling `cellprofiler --print-groups` from the command line (see [here](https://github.com/CellProfiler/CellProfiler/wiki/Adapting-CellProfiler-to-a-LIMS-environment#cmd) and [here](https://github.com/CellProfiler/Distributed-CellProfiler/issues/52) for more information).  
+  Note that for job files that specify groupings in this way, the `output_structure` variable is NOT optional - it must be specified or an error will be returned.
 
 ## Alternate job submission: run_batch_general.py
 
 We also support an alternate second path besides `submitJobs` to create the list of jobs - the `run_batch_general.py` file.
 This file essentially serves as a "shortcut" to run many common types of stereotyped experiments we run in our lab.
 Essentially, if your data follows a regular structure (such as N rows, N columns, N grouping, a particular structure for output, etc.), you may find it useful to take and modify this file for your own usage.  
 We recommend new users use the `submitJobs` pathway, as it will help users understand the kinds of information Distributed-CellProfiler needs in order to run properly, but once they are comfortable with it they may find `run_batch_general.py` helps them create jobs faster in the future.
+
+As of Distributed-CellProfiler 2.2.0, `run_batch_general.py` has been reformatted as a CLI tool with greatly enhanced customizeability.
+`run_batch_general.py` must be passed 5 pieces of information:
+
+### Required inputs
+
+* `step` is the step that you would like to make jobs for.
+Supported steps are `zproj`, `illum`, `qc`, `qc_persite`, `assaydev`, and`analysis`
+* `path_style` is the style of the input and output paths.
```

Comment:
What do you think about making this an optional parameter, with default as `default`? I know you personally do a lot of CPG processing but I don't know that I think that's going to be a general-enough phenomenon to want to make this flag mandatory on every run.

Reply:
(But I want to know what you think!)

Reply:
I guess it depends on what we think of as the purpose of run_batch_general.py. Is it our internal resource that we go ahead and make public or is it a public resource that we use a lot internally? If the former then `default` as default makes sense, but I guess I was thinking of it as the latter, especially as we've been both curating tools that access the CPG and pushing as much data to CPG as we can - in that case `cpg` as default (or keeping it as a required parameter) makes the most sense.